qpc/insights/utils.py

```diff
@@ -145,3 +145,19 @@ def validate_host(arg):
     if host_re.search(arg) is None:
         raise ArgumentTypeError(f"Host value {arg} should be a valid hostname")
     return arg
+
+
+def validate_username_and_password(arg):
+    """Validate password and username syntax.
+
+    :param arg: a string
+    :returns: The validated argument
+    :raises: ArgumentTypeError, if argument is invalid
+    """
+    argument_re = re.compile(
+        r"[A-Za-z0-9\!\"\#\$\%\&\'\(\)\*\+\,\-\.\/\:\;\<\>\=\?\@"
+        r"\[\]\{\}\\\\\^\_\`\~]+$"
```

Comment:
Oh, I think `r"^\S+$"` would achieve the desired results as well

Reply:
Or maybe if we want only ascii characters `r"[!-~]"`

Reply:
oh, that's much simpler. hehhe